### PR TITLE
Removed migration and vagrant driver

### PR DESCRIPTION
**Pull request by ssbarnea**
- Configuration migration from v1 to v2 is removed.
- Vagrant driver is removed, as this will be developed externally
  from https://github.com/pycontribs/molecule-vagrant

Fixes: #2315

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -132,7 +132,7 @@ ENV GEM_PACKAGES="\
     etc \
     "
 
-ENV MOLECULE_EXTRAS="azure,docker,docs,lxc,vagrant,windows"
```
Comment (decentral1se) [inline Dockerfile]: Banishing LXC once again :sweat_smile:

**Comment by Miouge1** [comment]
https://github.com/pycontribs/molecule-vagrant is an empty repo at the moment, and no molecule-vagrant project in pypi. Any idea where I can find the project?

**Comment by westurner** [comment]
Looks like these files didn't make it into https://github.com/ansible-community/molecule-vagrant :
- molecule/provisioner/ansible/playbooks/vagrant/create.yml
  https://github.com/ansible-community/molecule/blob/0cf231a3623e10456fbc21795ae42286ed88e479/molecule/provisioner/ansible/playbooks/vagrant/destroy.yml
- molecule/provisioner/ansible/playbooks/vagrant/create.yml
  https://github.com/ansible-community/molecule/blob/0cf231a3623e10456fbc21795ae42286ed88e479/molecule/provisioner/ansible/playbooks/vagrant/create.yml

I'm now getting the following errors:

```
--> Action: 'destroy'
ERROR! the playbook: None could not be found
ERROR: 
```
```
--> Action: 'create'
ERROR! the playbook: None could not be found
ERROR: 
```

[Where] should these be added in molecule-vagrant so that ``molecule init scenario -d vagrant -s fedora-vagrant`` works again?

**Comment by westurner** [comment]
Is this also an issue for other drivers that have been externalized for #2315 ?

**Comment by westurner** [comment]
And what is the last version before this change was released?

**Comment by westurner** [comment]
When I just copy create.yml and destroy.yml from the links above into the scenario directory, 

```bash
wget 'https://raw.githubusercontent.com/ansible-community/molecule/0cf231a3623e10456fbc21795ae42286ed88e479/molecule/provisioner/ansible/playbooks/vagrant/destroy.yml' 
wget 'https://raw.githubusercontent.com/ansible-community/molecule/0cf231a3623e10456fbc21795ae42286ed88e479/molecule/provisioner/ansible/playbooks/vagrant/create.yml' 
```

I'm now getting:
```
ERROR! couldn't resolve module/action 'molecule_vagrant'. This often indicates a misspelling, missing collection, or incorrect module path.
```
presumably because `molecule_vagrant.py` is not in https://github.com/ansible-community/molecule-vagrant ?
https://github.com/ansible-community/molecule/blob/0cf231a3623e10456fbc21795ae42286ed88e479/molecule/provisioner/ansible/plugins/modules/molecule_vagrant.py

https://github.com/ansible-community/molecule/pull/2414/files#diff-9b0cc6dd3c61a1aca9b9482e06c74b0f

**Pull request merged**